### PR TITLE
Upgrade Redis client to v9

### PR DESCRIPTION
**Pull request by trungdlp-wolffun**
Version 9.0.4 implements the support for Redis v7.

Changelog of redis/go-redis can be found here: https://github.com/redis/go-redis/blob/master/CHANGELOG.md

https://github.com/hibiken/asynq/pull/627

Fixed issue: https://github.com/hibiken/asynqmon/issues/302

### CODE REVIEW

**Comment by trungdlp-wolffun** [comment]
@kamikazechaser help me review this PR, please 😁

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Building a frontend.
 #
 
-FROM alpine:3.13 AS frontend
+FROM alpine:3.17 AS frontend
```
Comment (trungdlp-wolffun) [inline Dockerfile:7]: With OS security fix

**Comment by trungdlp-wolffun** [comment]
@hibiken 🥺

**Review by kamikazechaser — APPROVED** [review]
LGTM

**Comment by liangguifeng** [comment]
@hibiken I need this PR, and now asyncq @ latest and asyncqmon @ latest cannot work

**Review by lukasmalkmus — APPROVED** [review]

**Comment by trungdlp-wolffun** [comment]
@hibiken please review this PR 🚶

**Review by hibiken — APPROVED** [review]
LGTM! (sorry for the delay)

**Comment by lukasmalkmus** [comment]
> LGTM! (sorry for the delay)

No need to apologise! OSS maintainers have a life, too :)

I usually ping/ack/act on OSS issues/PRs because GitHub notifications can be overwhelming and are not very effective to manage.

**Pull request merged**